### DISCUSSION TITLE
if no WIFI or MQTT store data on Sd Card

### DISCUSSION THREAD

**Discussion by sdepascali** [arendst/Tasmota#18136] [Q&A]
Dear All,

I am very new on Tasmota world. I have created my own Device (Esp32 , some sensor) This Device will send data to MQTT Broker.
My question is how i can store data on SD Card if Wifi or MQTT are not working, but once Wifi or MQTT are online again send data from SD Card to  MQTT broker and delete data from SD Card. 

I this possible with Tasmota
Thank you in advance 
Salva

**Comment by sfromis** [comment]
Such buffering is not a feature of Tasmota. When you are using ESP32, you could create Berry code to collect data in a buffer (maybe in a file), to forward when there is a chance.

**Reply by barbudor** [reply]
@sdepascali
Alternatively on ESP8266 (and ESP32 too) you can use Scripting-Language
I've seen a few discussions on that topic lately
Use the search tool to find them

**Comment by sdepascali** [comment]
Hi 

Thank you i will try to research again

**Reply by wilriker** [reply]
Have you come across/implemented a solution? I came here looking for the same thing and wonder if you solved it.
